applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py

```diff
@@ -0,0 +1,474 @@
+                "solver_type" : "amgcl",
+                "smoother_type":"damped_jacobi",
+                "krylov_type": "cg",
+                "coarsening_type": "aggregation",
+                "max_iteration": 200,
+                "provide_coordinates": false,
+                "gmres_krylov_space_dimension": 100,
+                "verbosity" : 0,
+                "tolerance": 1e-7,
+                "scaling": false,
+                "block_size": 3,
+                "use_block_matrices_if_possible" : true,
+                "coarse_enough" : 50
+            }
+        }""")
+        this_defaults.AddMissingParameters(super(MPMSolver, cls).GetDefaultSettings())
+        return this_defaults
+
+    ### Solver public functions
+
+    def AddVariables(self):
+        # Add variables to background grid model part
+        self._add_variables_to_model_part(self.grid_model_part)
+
+        KratosMultiphysics.Logger.PrintInfo("::[MPMSolver]:: ", "Variables are added.")
+
+    def ImportModelPart(self):
+        # Read model part
+        self._model_part_reading()
+
+        KratosMultiphysics.Logger.PrintInfo("::[MPMSolver]:: ","Models are imported.")
+
+    def PrepareModelPart(self):
+        # Set buffer size
+        self._set_buffer_size()
+
+        # Executes the check and prepare model process
+        self._execute_check_and_prepare()
+
+        KratosMultiphysics.Logger.PrintInfo("::[MPMSolver]:: ", "ModelPart prepared for Solver.")
+
+    def GetComputingModelPart(self):
+        if not self.model.HasModelPart(self.settings["model_part_name"].GetString()):
+            raise Exception("The ComputingModelPart was not created yet!")
+        return self.model.GetModelPart(self.settings["model_part_name"].GetString())
+
+    def GetGridModelPart(self):
+        if not self.model.HasModelPart("Background_Grid"):
+            raise Exception("The GridModelPart was not created yet!")
+        return self.model.GetModelPart("Background_Grid")
+
+    def AddDofs(self):
+        # Add dofs to background grid model part
+        self._add_dofs_to_model_part(self.grid_model_part)
+
+        KratosMultiphysics.Logger.PrintInfo("::[MPMSolver]:: ","DOFs are added.")
+
+    def Initialize(self):
+        # The mechanical solution strategy is created here if it does not already exist.
+        particle_solution_strategy = self.get_solution_strategy()
+        particle_solution_strategy.SetEchoLevel(self.settings["echo_level"].GetInt())
+
+        # Generate material points
+        self.generate_material_point()
+
+        KratosMultiphysics.Logger.PrintInfo("::[MPMSolver]:: ","Solver is initialized correctly.")
+
+    def AdvanceInTime(self, current_time):
+        dt = self.ComputeDeltaTime()
+        new_time = current_time + dt
+        self.grid_model_part.ProcessInfo[KratosMultiphysics.STEP] += 1
+        self.grid_model_part.CloneTimeStep(new_time)
+
+        return new_time
+
+    def ComputeDeltaTime(self):
```

Comment:
```suggestion
    def __ComputeDeltaTime(self):
```
this is not a function in the interface, it should be private

Reply:
I will change it in upcoming pull request